pkg/graph/resolvers/system_intake_status_requester.go

```diff
@@ -0,0 +1,107 @@
+
+// CalculateSystemIntakeRequesterStatus calculates the status to display in the requester view for a System Intake request,
+// based on the intake's current step, the state of that step, and the overall intake state (open/closed)
+func CalculateSystemIntakeRequesterStatus(intake *models.SystemIntake, currentTime time.Time) (models.SystemIntakeStatusRequester, error) {
+	if intake.State == models.SystemIntakeStateCLOSED && intake.DecisionState == models.SIDSNoDecision {
+		return models.SISRClosed, nil
+	}
+
+	switch intake.Step {
+	case models.SystemIntakeStepINITIALFORM:
+		return calcSystemIntakeInitialFormStatusRequester(intake.RequestFormState)
+	case models.SystemIntakeStepDRAFTBIZCASE:
+		return calcSystemIntakeDraftBusinessCaseStatusRequester(intake.DraftBusinessCaseState)
+	case models.SystemIntakeStepGRTMEETING:
+		// this calc function doesn't use a switch statement and can't possibly return an error
+		return calcSystemIntakeGRTMeetingStatusRequester(intake.GRTDate, currentTime), nil
+	case models.SystemIntakeStepFINALBIZCASE:
+		return calcSystemIntakeFinalBusinessCaseStatusRequester(intake.FinalBusinessCaseState)
+	case models.SystemIntakeStepGRBMEETING:
+		// this calc function doesn't use a switch statement and can't possibly return an error
+		return calcSystemIntakeGRBMeetingStatusRequester(intake.GRBDate, currentTime), nil
+	case models.SystemIntakeStepDECISION:
+		return calcSystemIntakeDecisionStatusRequester(intake.DecisionState)
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid step")
+	}
+}
+
+func calcSystemIntakeInitialFormStatusRequester(intakeFormState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch intakeFormState {
+	case models.SIRFSNotStarted:
+		return models.SISRInitialRequestFormNew, nil
+	case models.SIRFSInProgress:
+		return models.SISRInitialRequestFormInProgress, nil
+	case models.SIRFSEditsRequested:
+		return models.SISRInitialRequestFormEditsRequested, nil
+	case models.SIRFSSubmitted:
+		return models.SISRInitialRequestFormSubmitted, nil
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid intakeFormState")
+	}
+}
+
+func calcSystemIntakeDraftBusinessCaseStatusRequester(draftBusinessCaseState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch draftBusinessCaseState {
+	case models.SIRFSNotStarted, models.SIRFSInProgress:
+		return models.SISRDraftBusinessCaseInProgress, nil
+	case models.SIRFSEditsRequested:
+		return models.SISRDraftBusinessCaseEditsRequested, nil
+	case models.SIRFSSubmitted:
+		return models.SISRDraftBusinessCaseSubmitted, nil
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid draftBusinessCaseState")
+	}
+}
+
+func calcSystemIntakeGRTMeetingStatusRequester(grtDate *time.Time, currentTime time.Time) models.SystemIntakeStatusRequester {
+	if grtDate == nil || grtDate.After(currentTime) {
```

Comment:
Shouldn't this status have a `READY_TO_SCHEDULE` for no date, `SCHEDULED` for a future date, and `AWAITING_DECISION` for a past date?

Reply:
Not for the overall intake status - that only has "Ready for GRT meeting" and "Awaiting GRT decision" in the requester view. ([Figma](https://www.figma.com/file/ChzAP34A2DVvQUNQwD7lCt/IT-Governance-Next?type=design&node-id=475-115531&mode=design&t=foWugeQRbZN35Xtx-0)) The task list status for the GRT step will have the different values you're mentioning, that'll be covered in EASI-2960.

Reply:
That's weird, but OK, if that's what the designs call for. Thanks for clarifying.